datalad/utils.py

```diff
@@ -207,7 +207,11 @@ def is_interactive():
     """Return True if all in/outs are tty"""
     # TODO: check on windows if hasattr check would work correctly and add value:
     #
-    return sys.stdin.isatty() and sys.stdout.isatty() and sys.stderr.isatty()
+    return (
+        (not sys.stdin.closed and sys.stdin.isatty()) and
+        (not sys.stdout.closed and sys.stdout.isatty()) and
+        (not sys.stderr.closed and sys.stderr.isatty())
```

Comment:
oh hoh, `.closed` itself pukes if stdout is closed
```shell
$> python2 /tmp/1.py 
Traceback (most recent call last):
  File "/tmp/1.py", line 3, in <module>
    print(sys.stdout.closed)
ValueError: I/O operation on closed file

$> python3 /tmp/1.py 
Traceback (most recent call last):
  File "/tmp/1.py", line 3, in <module>
    print(sys.stdout.closed)
ValueError: I/O operation on closed file.

$> cat /tmp/1.py    
import sys; 
sys.stdout.close(); 
print(sys.stdout.closed)
```
so I think there is no way around try/except :-/